File: 0763-partition-labels/0763-partition-labels.cs
public class Solution {
    public IList<int> PartitionLabels(string s) {
        var n = s.Length;

        // store last occurence of a character
        var last = new int[26];
        for(var i=0; i<n; i++){
            var idx = s[i] - 'a';
            last[idx] = i;
        }

        var ans = new List<int>();
        int start = 0, end = 0;
        for(var i=0; i<n; i++){
            var idx = s[i] - 'a';
            if(end < i) {
                ans.Add(end - start + 1);
                start = i;
            }
            end = Math.Max(end, last[idx]);    
        }

        ans.Add(end - start + 1);
        return ans;
    }
}

File: 0763-partition-labels/0763-partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        n = len(s)

        # store last occurence of a character
        last = [0] * 26
        for i, ch in enumerate(s):
            idx = ord(ch) - ord('a')
            last[idx] = i

        ans = []
        start = end = 0
        for i, ch in enumerate(s):
            idx = ord(ch) - ord('a')
            if end < i:
                ans.append(end - start + 1)
                start = i
            end = max(end, last[idx])

        ans.append(end - start + 1)
        return ans